src/modules/launcher/Plugins/Microsoft.Plugin.Program/Main.cs

```diff
@@ -64,26 +66,19 @@ public Main()
             Task.WaitAll(a, b);
 
             _settings.LastIndexTime = DateTime.Today;
+            
         }
 
         public void Save()
         {
             _settingsStorage.Save();
-            _win32Storage.Save(_win32s);
+            _win32ProgramRepository.Save();
             _packageRepository.Save();
         }
 
         public List<Result> Query(Query query)
         {
-            Programs.Win32[] win32;
-
-            lock (IndexLock)
-            {
-                // just take the reference inside the lock to eliminate query time issues.
-                win32 = _win32s;
-            }
-
-            var results1 = win32.AsParallel()
+            var results1 = _win32ProgramRepository.AsParallel()

```

Comment:
I think there would be a synchronization issue here in case a program is installing/uninstalling and a query is made. The earlier code was handling this.

Reply:
We are using a ConcurrentDictionary which is threadsafe. I don't think this should be an issue. @ryanbodrug-microsoft, do you think there would be an issue?

Reply:
My bad. That should be fine. @ryanbodrug-microsoft can confirm.